src/App.js

```diff
@@ -1,65 +1,91 @@
 import "./styles.css";
 import { useEffect } from "react";
-import { ProductListing } from "./Product-listing";
-import { Cart } from "./Cart";
-import { Wishlist } from "./Wishlist";
-import { serverRequest } from "./server-request";
-import { useStateContext } from "./context";
-import { Nav } from "./Header";
-import { Home } from "./Home";
-import { Footer } from "./Footer";
-import { ForgotPassword, Login } from "./Authentication";
+import { useStateContext, useAuthentication } from "./context";
+import { getProductsFromServer } from "./utils";
 import { Routes, Route } from "react-router-dom";
-import { PrivateRoute } from "./Authentication";
-import { Profile } from "./Authentication";
-import { SignUp } from "./Authentication/Sign-up";
+
+import {
+  ProductListing,
+  Cart,
+  Wishlist,
+  Nav,
+  Home,
+  Footer,
+  ForgotPasswordPage,
+  Login,
+  PrivateRoute,
+  Profile,
+  SignUp,
+  ProductDetailPage,
+  ErrorPage
+} from "./Components";
 
 export default function App() {
   const { dispatch } = useStateContext();
+  const { isUserLoggedIn, userId } = useAuthentication();
 
   useEffect(() => {
     (async () => {
       try {
-        const { response } = await serverRequest({
-          url: "api/products",
-          requestType: "GET"
-        });
-        dispatch({ type: "SET_PRODUCTS", payload: response.data.products });
-      } catch {}
-      try {
-        const { response } = await serverRequest({
-          url: "api/carts",
-          requestType: "GET"
-        });
-        dispatch({ type: "SET_CART", payload: response.data.carts });
-      } catch {}
-      try {
-        const { response } = await serverRequest({
-          url: "api/wishlists",
-          requestType: "GET"
-        });
-
-        dispatch({
-          type: "SET_WISHLIST",
-          payload: response.data.wishlists
-        });
-      } catch {}
+        const {
+          data: { response }
+        } = await getProductsFromServer(
+          "https://uandistoreapi.herokuapp.com/products"
+        );
+        dispatch({ type: "SET_PRODUCTS", payload: response });
+      } catch (error) {
+        console.log(error);
```

Comment:
remove console from here. You can use here notification (toastify) to send error to user